### PR TITLE
Add Dropdown component

### PR DESCRIPTION
**Pull request by euanlau**
[#161328391]

### CODE REVIEW

File: tests/Tooltip.test.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import React from 'react';
-import { shallow } from 'enzyme';
-
+import mountWithTheme from './utils/mountWithTheme';
 import Tooltip from '../src/Tooltip';
 
 describe('<Tooltip />', () => {
   it('properly renders a tooltip', () => {
-    const tree = shallow(
+    const tree = mountWithTheme(
```
Comment (euanlau) [inline tests/Tooltip.test.js]: I couldn't find anything that says `jest-emotion` (https://github.com/emotion-js/emotion/tree/master/packages/jest-emotion) actually works with `shallow`.  
`jest-enzyme` (which i added) does have support for that however. Adding that package on the other hand breaks the shallow snapshot. Reason unknown. I assume it is due to the fact that it requires Theme?
So I use mount here instead.

**Comment by dgalarza** [comment]
Dropdown is looking good.

Couple things I noticed though:

1. It looks like the `Menu` storybook story is not really necessary anymore. WDUT?
2. After you select a menu item in the dropdown and re-open it, the menu item is no longer selected. Is that intentional? Do we want that handled elsewhere? If we do, we probably need to support passing in which item is selected down into the menu. I originally left a note on the pivotal story that Menu has some state right now for this but I'm not sure it's appropriate. It was sort of a placeholder to handle selected states before.

**Comment by euanlau** [comment]
1. `Menu` storybook should just be a menu instead of a combination of `Popover` and `Menu`?
2. I planned to utilize the `selected` prop (which I forgot to add) since the menu will be remounted when the popover reopens.

**Comment by euanlau** [comment]
@dgalarza addressed the two things that you mentioned.

File: src/Menu/index.js [hunk 2]
```diff
@@ -150,11 +150,17 @@ Menu.propTypes = {
       name: PropTypes.string,
       value: PropTypes.string
     })
-  ).isRequired
+  ).isRequired,
+
+  /**
+   * Initial selected menu item value
+   */
+  selected: PropTypes.string
```
Comment (dgalarza) [inline src/Menu/index.js]: @euanlau we probably want to remove the `state` `selected` property now that we're passing in selected as a prop and update `renderMenuItem` appropriately.

Comment (dgalarza) [reply src/Menu/index.js]: Also, it mentions that `selected` is a string and the selected menu item. Does this represent the value of the menu item? The ID? We should probably update the doc here to be more specific. I imagine once the state is removed from the component, the test will also help identify what we mean here.

**Comment by dgalarza** [comment]
> Menu storybook should just be a menu instead of a combination of Popover and Menu?

Works for me.

File: src/Menu/index.js (hunk 2 above)
Comment (euanlau) [reply src/Menu/index.js]: That is modifying the behavior of Menu itself is out of the scope of this story I think?

Comment (euanlau) [reply src/Menu/index.js]: I just added the `PropTypes` since it was missing.

**Pull request merged**